src/main/java/edu/ie3/datamodel/io/source/sql/SqlWeatherSource.java

```diff
@@ -361,4 +361,9 @@ private String createQueryStringForTimeIntervalAndCoordinates(
     }
     return coordinateToTimeSeries;
   }
+
+  private Connection getConnection() throws SQLException {
```

Comment:
From my POV we should discuss, if holding a connection should better be moved to the connector itself. Either `SqlConnector` having an attribute `reuseConnection` or the getter being like

```java
public getConnection() throws SQLException {
  return getConnection(false);
}

public getConnection(boolean reuseConnection) throws SQLException {
  if (reuseConnection && (connection == null || connection.isClosed()) || !reuseConnection) connection = connector.getConnection();
    return connection;
}
```

What are your thoughts on this?

Reply:
Very good point. Especially bc otherwise it would be impossible to execute concurrent db calls with the same weather source (we would have to create a new weather source for each concurrent call). However, I would prefer having a slightly more elaborated approach, where the number of parallel connections is tracked in order to not create too much concurrent connections. This might be possible with a list or map holding the actual open connections and either returns one of them that is currently not used or creates a new one. Furthermore, in this case the connector should keep "its space clean" by automatically remove unused connections whenever a new `getConnection` call is executed.

Maybe we can discuss this in a short call?

Reply:
![](https://media.giphy.com/media/dZQR7P1lxgi4mUmsjY/giphy.gif)

I'm open to a call. But then we might consider splitting up the PR into on InfluxDB- and one SQL-PR?!